qir/qat/ModuleLoader/ModuleLoader.hpp

```diff
@@ -38,33 +42,74 @@ namespace quantum
             return !linker_.linkInModule(std::move(module), Linker::Flags::None);
         }
 
-        bool addIrFile(String const& filename)
+        bool addIrFile(String input_file)
         {
+            // Converting to absolute path
+            llvm::SmallVector<char, 256> input_vec;
+            input_vec.assign(input_file.begin(), input_file.end());
+            llvm::sys::fs::make_absolute(input_vec);
+            input_file.resize(input_vec.size());
+            uint64_t i = 0;
+            for (auto& s : input_vec)
+            {
+                input_file[i++] = s;
+            }
 
             // Loading module
             SMDiagnostic            err;
-            std::unique_ptr<Module> module = llvm::parseIRFile(filename, err, final_module_->getContext());
+            std::unique_ptr<Module> module = llvm::parseIRFile(input_file, err, final_module_->getContext());
             if (!module)
             {
-                llvm::errs() << "Failed to load " << filename << "\n";
+                llvm::errs() << "Failed to load " << input_file << "\n";
                 return false;
             }
 
+            // Registering all debug info
+            auto directory = llvm::sys::path::parent_path(input_file);
+            auto filename  = llvm::sys::path::filename(input_file);
+
+            debug_table_->registerModule(input_file, module.get());
+
+            if (strip_existing_debug_)
+            {
+                // Debug info
+                llvm::StripDebugInfo(*module.get());
+            }
+
+            // Whether or not to override debug symbols
+            if (add_ir_debug_info_)
+            {
+
+                // Adding debug versioning
+                auto debug_version_key = "Debug Info Version";
+                if (!module->getModuleFlag(debug_version_key))
+                {
+                    module->addModuleFlag(llvm::Module::Warning, debug_version_key, llvm::DEBUG_METADATA_VERSION);
+                }
+
+                // Update with debug information
+                DebugInfoUpdater updater(debug_table_, *module.get(), directory, filename);
+                updater.update();
+            }
```

Comment:
I'm a little confused by the intent of `strip_existing_debug_` here. The existing debug information is unconditionally captured into the table, then the info is optionally stripped from the module. This means the strip option doesn't suppress the existing debug info from being present during later feedback. So is the debug table and it's use meant to be independent of the strip option or could there be an ordering issue here?

Reply:
`strip_existing_debug_` refers to the debug symbols that existed _before_ the IR was loaded by QAT. Notice that interchanging the strip-if block and line 71 does not change the meaning of the program. I think the confusion is that `debug_table_` really is a `location_table_`. I will change the naming.